### PR TITLE
Ember 4.0 compatibility - drops support for Ember < 3.28

### LINKED ISSUE #701: Installing this addon breaks build on Ember 4
### Environment

- Ember Version: 4.0.1
- Ember CLI Version: 4.0.1
- Ember CP Validations Version: 4.0.0

### Steps to Reproduce
Create a brand new ember app on the latest Ember and install ember-cp-validations addon. Run ember build. It will break with the error:

WARNING: [DEPRECATION] [DEPRECATION] Usage of the Ember Global is deprecated. You should import the Ember module or the specific API instead.

See https://deprecations.emberjs.com/v3.x/#toc_ember-global for details.

Usages of the Ember Global may be caused by an outdated ember-cli-babel dependency. The following steps may help:

* Upgrade the following addons to the latest version:
  * ember-cp-validations

### Important ###

In order to avoid repeatedly showing the same deprecation messages, no further deprecation messages will be shown for usages of the Ember Global until ember-cli-babel is upgraded to v7.26.6 or above.

To see all instances of this deprecation message, set the `EMBER_GLOBAL_DEPRECATIONS` environment variable to "all", e.g. `EMBER_GLOBAL_DEPRECATIONS=all ember test`.

### Details ###

Prior to v7.26.6, ember-cli-babel sometimes transpiled imports into the equivalent Ember Global API, potentially triggering this deprecation message indirectly, even when you did not observe these deprecated usages in your code.

The following outdated versions are found in your project:

* ember-cli-babel@6.18.0, currently used by:
  * ember-require-module@0.3.0
    * Depends on ember-cli-babel@^6.9.2
    * Added by ember-cp-validations@4.0.0-beta.12
    * Added by ember-cp-validations@4.0.0-beta.12 > ember-validators@3.0.1
  * ember-validators@3.0.1
    * Depends on ember-cli-babel@^6.9.2
    * Added by ember-cp-validations@4.0.0-beta.12

Environment: development
cleaning up...
Template Compiler Error (TemplateCompiler) in ember-quickstart/templates/application.hbs

Cannot read properties of undefined (reading 'visitor')

Stack Trace and Error Report: /var/folders/7l/54hcgmcj5px90x_b3n2lkss00000gn/T/error.dump.230e7976d87f3c46c3e6cfa1b4b3edcb.log

### PR DESCRIPTION
**Pull request by fsmanuel**
Breaking changes:
- Drop support for Ember < 3.28

First attempt to make it ember 4.0 compatible.

- [x] Remove `volatile()` from the code base
- [x] https://github.com/qonto/ember-cp-validations/pull/13

Thanks to @zeppelin and @qonto I could cherry pick a lot from the fork.

ref #716 
closes #701

### CODE REVIEW

**Review by tehhowch — COMMENTED** [review]
I don't have an ember 3.28/4+ app to use to verify this PR remains / fix compatibility. Didn't look at changes to tests or lines not modified, either (i.e. not a substantiative review).

**Review by MelSumner — APPROVED** [review]
Let's move forward with this. 👍

**Comment by tehhowch** [comment]
@fsmanuel i was able to pull this down and run unit & integration tests for some just-migrated-to-3.28 apps, without issues. We have more extensive test suites but they are more costly to run, so they'll need to wait until this PR is released in 5.0

I'm curious: does this PR drop support for Ember 3.24 just because, or are there things only in 3.28 or later that are necessary for the current implementation?

**Comment by kiwi-josh** [comment]
As another data point (as someone running a large ember app in production on the https://github.com/qonto/ember-cp-validations/commit/3767e4114e0c327138287486094e228bd71a45ec fork).
I pointed our repo at this PR, smoke tested a bunch of validation functionality, and ran our test suite, and all green 😄

**Comment by fsmanuel** [comment]
@MelSumner thanks for approving!
@tehhowch @kiwi-josh thanks for testing with a real app.

@tehhowch I think I needed to drop support for 3.24 but I'm not 100% sure. The general problem with old versions is that the maintaining efforts for the test coverage increases a lot. For example ember bootstrap was used in this addon for the docs and then you need to figure out which ember, addon and ember bootstrap versions work well together.
Because ember 4 is significantly different it's best to drop the test coverage for the old versions to make life easier with the upcoming changes to this addon.

File: addon/validations/factory.js [hunk 1]
```diff
@@ -812,33 +796,30 @@ function resolveDebounce(resolve) {
...
+    }
 
-        // If an async validation is found, throw an error
-        if (!isAsync && get(validationResult, 'isAsync')) {
-          throw new Error(
-            `[ember-cp-validations] Synchronous validation failed due to ${name} being an async validation.`
-          );
-        }
+    if (isEmpty(whiteList) || whiteList.indexOf(name) !== -1) {
+      let validationResult = get(this, `attrs.${name}`);
```
[the remark this answers is not in the document]
Comment (fsmanuel) [reply addon/validations/factory.js:809]: For now it is.

**Comment by fsmanuel** [comment]
Released as `5.0`

**Pull request merged**